artemis-protocols/artemis-openwire-protocol/src/main/java/org/apache/activemq/artemis/core/protocol/openwire/OpenWireMessageConverter.java

```diff
@@ -927,6 +932,10 @@ private static void setAMQMsgObjectProperties(final ActiveMQMessage amqMsg,
                                                  final Set<SimpleString> props) throws IOException {
       for (SimpleString s : props) {
          final String keyStr = s.toString();
+         if (keyStr.length() == 0) {
```

Comment:
shouldn't it check for keyStr.trim().isEmpty() instead?


If I add a property named "   " it will still go through and be converted.


as a matter of fact the test could add such condition?

Reply:
Apparently whitespace is legal. I tested it and it passed without any exception.